pkg/agent/route/route_linux.go

```diff
@@ -152,7 +152,55 @@ func (c *Client) syncIPInfra() {
 		klog.Errorf("Failed to sync iptables: %v", err)
 		return
 	}
-	klog.V(3).Infof("Successfully synced node iptables")
+	if err := c.syncRoutes(); err != nil {
+		klog.Errorf("Failed to sync routes: %v", err)
+	}
+	klog.V(3).Infof("Successfully synced node iptables and routes")
+}
+
+func (c *Client) syncRoutes() error {
+	routeList, err := netlink.RouteList(nil, netlink.FAMILY_ALL)
```

Comment:
There might be race condition between `syncRoutes` and `DeleteRoutes`: 
if `DeleteRoutes` has deleted actual routes and is about to delete the desired routes, `syncRoutes` may see that there are desired rules but no actual routes and add the routes back. Then the routes will going to be stale.